packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -84,6 +84,9 @@ export abstract class CredentialService<CFs extends CredentialFormat[] = Credent
   // methods for ack
   abstract processAck(messageContext: InboundMessageContext<AgentMessage>): Promise<CredentialExchangeRecord>
 
+  // methods for problem-report
+  abstract createProblemReport(message: string): ProblemReportMessage
```

Comment:
Same here, if we make it an object it would be easier to add new properties later on (such as exposing the `code`)

Reply:
refactored